includes/class-amp-theme-support.php

```diff
@@ -1732,12 +1756,18 @@ public static function prepare_response( $response, $args = array() ) {
 					);
 				}
 
+				$response = esc_html__( 'Redirecting to non-AMP version.', 'amp' );
+
+				if ( $cache_response ) {
```

Comment:
Good idea to cache the response when there are blocking errors and it has to redirect to a non-AMP URL.